Kusto/schema migrations

hudo opened the pull request.

VaclavK commented on src/Eshopworld.Telemetry/Kusto/KustoOptionsBuilder.cs:
```diff
@@ -2,126 +2,156 @@
-            types.Add(typeof(T));
+            if (_optionsSet)
+            {
+                throw new InvalidOperationException("You can only set options once, and this Kusto Options Builder already has options set.");
```
would remove this as not functional .. this can be covered via doco

hudo replied:
agree

hudo replied:
Its a signature of IBB from ESW.Core, will remove this and update Core package

djfr commented:
@hudo I have no idea why did you PR one of my working branches … nor why would you PR if it wasn't working …

I fixed the `CslType` issue in the `ColumnSchema` and added a test that migrates twice a new type of test event `KustoMigrationTestEvent`.

This could still use more tests, but since it's your PR, feel free to add them.

I'm not adding myself as a review to this PR, since reviewing my own code doesn't feel right, so if you need me to chip in, include me in a comment.

hudo commented:
> 
> 
> @hudo I have no idea why did you PR one of my working branches … nor why would you PR if it wasn't working …
> 
> I fixed the `CslType` issue in the `ColumnSchema` and added a test that migrates twice a new type of test event `KustoMigrationTestEvent`.
> 
> This could still use more tests, but since it's your PR, feel free to add them.
> 
> I'm not adding myself as a review to this PR, since reviewing my own code doesn't feel right, so if you need me to chip in, include me in a comment.

You said you are done, and I just tried to fix it because it wasn't working.
Closed this PR and will work on separate branch without migrations to get new Kusto implementation ready.
